src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -42,39 +43,85 @@ public async Task<(Error, GitHubUser)> GetUserByLogin(string login)
                 return (_rateLimitError, null);
             }
 
-            var apiDetail = $"GET /users/{login}";
+            if (_unauthorizedError != null)
+            {
+                return default;
+            }
+
+            var query = @"
+query ($login: String!) {
+  user(login: $login) {
+    name
+    email
+    databaseId
+  }
+}";
+
             try
             {
-                var user = await RetryUtility.Retry(
-                    () => _client.User.Get(login),
-                    ex => ex is OperationCanceledException);
+                var (error, response) = await Query(query, new
+                {
+                    login,
+                });
+
+                if (error != null)
+                {
+                    return (error, null);
+                }
+
+                var grahpApiResponse = JsonConvert.DeserializeAnonymousType(await response.Content.ReadAsStringAsync(), new
+                {
```

Comment:
Concrete the type definition into a strong type model? Feels a bit too long here and not very straightforward.

Reply:
there are too many depths of Grahp API Result, and most of them are not re-usable,  so I would like use anonymous type,

Reply:
how about just loading the string into JToken? what is the benefit of another de-serialization into anonymous type?